src/rpc/misc.cpp

```diff
@@ -158,11 +158,8 @@ UniValue validateaddress(const JSONRPCRequest& request)
 
 #ifdef ENABLE_WALLET
     CWallet * const pwallet = GetWalletForJSONRPCRequest(request);
-
-    LOCK2(cs_main, pwallet ? &pwallet->cs_wallet : nullptr);
-#else
-    LOCK(cs_main);
 #endif
+    LOCK(cs_main);
```

Comment:
@luke-jr but agree with this change?

Reply:
Does IsMine not require the lock? If you're going to do this, I suggest moving `GetWalletForJSONRPCRequest` closer to the lock as well.

Reply:
> Does IsMine not require the lock

I don't think so.